### PR TITLE
fix creating record file when not required

### PR DESCRIPTION
**Pull request by wmartinmimi**
Fixes: #72

### CODE REVIEW

File: lua/typst-preview/fetch.lua [hunk 1]
```diff
@@ -209,14 +209,22 @@ function M.fetch(quiet, callback)
...
       )
     end
-    local record, err = io.open(record_path, 'w')
-    if record == nil then
-      error("Can't open record file!: " .. err)
-    end
+    local record_required = true
     for _, bin in pairs(M.bins_to_fetch()) do
-      record:write(bin.url .. '\n')
+      if config.opts.dependencies_bin[bin.name] ~= nil then
```
Comment (chomosuke) [inline lua/typst-preview/fetch.lua:214]: Wouldn't this mark record_required as false when one of the dependencies_bin doesn't need to be downloaded but the other one does?

File: lua/typst-preview/fetch.lua [hunk 2]
```diff
@@ -209,14 +209,22 @@ function M.fetch(quiet, callback)
...
+        record_required = false
+      end
+    end
+    if record_required then
+      local record, err = io.open(record_path, 'w')
+      if record == nil then
+        error("Can't open record file!: " .. err)
+      end
+      for _, bin in pairs(M.bins_to_fetch()) do
```
Comment (chomosuke) [inline lua/typst-preview/fetch.lua:223]: Perhaps it's best to only write the bin that are not in dependencies_bin into record file.

**Review by chomosuke — CHANGES_REQUESTED** [review]
Thank you for this PR!

**Review by chomosuke — APPROVED** [review]
Now it still creates `version_record.txt` but it doesn't error since the parent directories will be created

File: lua/typst-preview/fetch.lua (hunk 1 above)
Comment (wmartinmimi) [reply lua/typst-preview/fetch.lua]: ah true, my bad

File: lua/typst-preview/fetch.lua (hunk 2 above)
Comment (wmartinmimi) [reply lua/typst-preview/fetch.lua]: yes, that would be better

**Pull request merged**